vortex-serde/src/layouts/pruning.rs

```diff
@@ -0,0 +1,538 @@
+use std::collections::hash_map::Entry;
+use std::sync::Arc;
+
+use ahash::{HashMap, HashMapExt};
+use vortex::stats::Stat;
+use vortex_dtype::field::Field;
+use vortex_dtype::Nullability;
+use vortex_expr::{BinaryExpr, Column, Literal, Operator, VortexExpr};
+use vortex_scalar::Scalar;
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct ColumnStat {
+    column: Field,
+    stat: Stat,
+}
+
+impl ColumnStat {
+    pub fn new(column: Field, stat: Stat) -> Self {
+        Self { column, stat }
+    }
+
+    pub fn column(&self) -> &Field {
+        &self.column
+    }
+
+    pub fn stat(&self) -> Stat {
+        self.stat
+    }
+}
+
+#[allow(dead_code)]
+pub struct PruningPredicate {
+    expr: Arc<dyn VortexExpr>,
+    stat_references: HashMap<Field, ColumnStat>,
```

Comment:
Don't we actually store every `field` name twice here? Once as the key and once in the `ColumnStat` value